x-pack/plugins/encrypted_saved_objects/server/saved_objects/encrypted_saved_objects_client_wrapper.ts

```diff
@@ -212,13 +212,20 @@ export class EncryptedSavedObjectsClientWrapper implements SavedObjectsClientCon
       options?.namespace
     );
     return this.handleEncryptedAttributesInResponse(
-      await this.options.baseClient.update(
+      await this.options.baseClient.create<T>(
         type,
-        id,
-        await this.options.service.encryptAttributes({ type, id, namespace }, attributes, {
-          user: this.options.getCurrentUser(),
-        }),
-        options
+        (await this.options.service.encryptAttributes(
+          { type, id, namespace },
+          attributes as Record<string, unknown>,
+          {
+            user: this.options.getCurrentUser(),
+          }
+        )) as T,
+        {
+          id,
+          overwrite: true,
+          ...options,
```

Comment:
>Also SavedObjectsUpdateOptions has version property that create doesn't understand. It's doesn't matter right now, but If SO client suddenly starts validating arguments at some point it may start throwing exceptions because of unknown arguments

We've refrained from doing this type of change to ESO before because of the optimistic concurrency control that the alerts currently depend on (passing the version into the update statement and fail if the alert changed since it was last read). I don't think it would be a good idea to let that go as we rely more and more on optimistic concurrency control for updating new attributes down the line (state, status, etc).

Most problems with AAD are planned to be fixed with custom code in this issue: https://github.com/elastic/kibana/issues/67290#issuecomment-638799954. It seems another case that approach would solve.

Reply:
Regarding the order - fair, I'll flip them. 👍 

Regarding SO - indeed, and I've already spoken to Platform as I'm, adding version into their type for this exact reason, otherwise we lose our optimistic concurrency control 👍